python/pyspark/ml/classification.py

```diff
@@ -1423,7 +1425,18 @@ def _fit(self, dataset):
 
         numClasses = int(dataset.agg({labelCol: "max"}).head()["max("+labelCol+")"]) + 1
 
-        multiclassLabeled = dataset.select(labelCol, featuresCol)
+        weightCol = None
+        if (self.isDefined(self.weightCol) and self.getWeightCol()):
+            if isinstance(classifier, HasWeightCol):
+                weightCol = self.getWeightCol()
+            else:
+                warnings.warn("weightCol is ignored, "
+                              "as it is not supported by {} now.".format(classifier))
```

Comment:
To pass Python2.6 test, we need to change this line to:
```
"as it is not supported by {0} now.".format(classifier))
```
FYI https://stackoverflow.com/questions/21034954/valueerror-zero-length-field-name-in-format-in-python2-6-6